File: Core/Team/src/main/java/org/team/application/commands/CreateTeam.java
package org.team.application.commands;

import org.shared.api.Command;
import org.team.entities.Team;
import org.team.ports.dto.CreateTeamRequest;
import org.team.ports.spi.TeamRepository;


public class CreateTeam implements Command<CreateTeamRequest> {
  private final TeamRepository teamRepository;

  public CreateTeam(TeamRepository teamRepository) {
    this.teamRepository = teamRepository;
  }

  public void execute(CreateTeamRequest request) {
    var team = new Team(request.id(), request.name(), request.teammates(), request.accountId());
    teamRepository.saveTeam(team.snapshot());
  }
}

File: frontend/src/core/application/team/command/CreateTeam.ts
import { Observable, catchError, map } from "rxjs";
import { CreateTeamFormData, CreateTeamRequest } from "../dto/CreateTeamFormData";
import { TeamCommandHandler } from "../spi/TeamCommandHandler";
import { Result } from "src/core/application/shared/dto/Result";
import { IDProvider } from "src/core/application/shared/interfaces/IDProvider";
import { Command } from "src/core/application/shared/command/Command";
import { TeamCard } from "src/core/model/team/TeamCard";


export class CreateTeam extends Command<TeamCard, CreateTeamFormData> {

    public constructor(private commandHandler: TeamCommandHandler, 
                       private idProvider: IDProvider){super()}

    public execute(formData: CreateTeamFormData): Observable<Result<TeamCard>> {
        const request: CreateTeamRequest = {...formData, id: this.idProvider.generate()}
        return this.commandHandler.createTeam(request)
        .pipe(map(this.onSuccess), 
        catchError(this.onError));
    }
}